src/main/java/emissary/core/channels/FileChannelFactory.java

```diff
@@ -82,6 +82,8 @@ protected void closeImpl() throws IOException {
         @Override
         protected int readImpl(ByteBuffer byteBuffer, int maxBytesToRead) throws IOException {
             initialiseChannel();
+            // Set position for underlying channel, otherwise this won't match the outer channel position
```

Comment:
It seems odd to be setting the FileChannel's position here, rather than setting it as part of the `position(final long position)` call.

Reply:
Do you mean as in overriding the `position(final long position)` method?

Reply:
Well, that the `position(final long position)` method in `AbstractSeekableByteChannel` is marked as `final`, but it might be reasonable to have that method to call an optional `positionImpl(final long position)` method that channel implementations could override as needed.

Reply:
Gotcha. I think given this is the only implementation currently, I'll hang fire until we have another one and do the refactoring at that point. Agree that having a more extensible/override-able base class would be desirable in the future.